doc/introduction/compiling_circuits.rst

```diff
@@ -211,77 +211,35 @@ For more details on :func:`~.pennylane.compile` and the available compilation tr
-    original_qnode = qml.QNode(circuit, original_dev)
-
->>> weights = np.array([[0.4, 0.5, 0.6]])
->>> print(qml.draw(original_qnode, level="device")(weights))
-0: ──RX(0.40)─╭●────╭X─┤  <Z>
-1: ──RX(0.50)─╰X─╭●─│──┤     
-2: ──RX(0.60)────╰X─╰●─┤     
-
-Now, let's swap out PennyLane's default decomposition of the ``CNOT`` gate into ``CZ``
-and ``Hadamard``.
-We define the custom decompositions like so, and pass them to a device:
-
-.. code-block:: python
-
-    def custom_cnot(wires, **_):
-        return [
-            qml.Hadamard(wires=wires[1]),
-            qml.CZ(wires=[wires[0], wires[1]]),
-            qml.Hadamard(wires=wires[1])
-        ]
+Gate decompositions
+-------------------
 
-    custom_decomps = {qml.CNOT: custom_cnot}
-
-    decomp_dev = qml.device("default.qubit", wires=3, custom_decomps=custom_decomps)
-    decomp_qnode = qml.QNode(circuit, decomp_dev)
-
-Note that custom decomposition functions should accept keyword arguments even when it is not used.
-
-Now when we draw or run a QNode on this device, the gates will be expanded
-according to our specifications:
-
->>> print(qml.draw(decomp_qnode, level="device")(weights))
-0: ──RX(0.40)────╭●──H───────╭Z──H─┤  <Z>
-1: ──RX(0.50)──H─╰Z──H─╭●────│─────┤     
-2: ──RX(0.60)──H───────╰Z──H─╰●────┤     
+When compiling a circuit, it is often beneficial to decompose the circuit into a 
+set of gates. To do this, we can use the :func:`~.pennylane.transforms.decompose` 
+function, which enables decomposition of circuits into a set of gates defined either 
+by their name, type, or by a set of rules they must follow.
 
 .. note::
-    If the custom decomposition is only supposed to be used in a specific code context,
-    a separate context manager :func:`~.pennylane.transforms.set_decomposition` can be used.
-
-Circuit Decomposition
----------------------
 
-When compiling a circuit it is often beneficial to decompose the circuit into a set of basis gates.  
-To do this, we can use the :func:`~.pennylane.transforms.decompose` function, which enables decomposition of
-circuits into a set of gates defined either by their name, type, or by a set of rules they must follow.
+    Using :func:`~.pennylane.decompositions.enable_graph` enables PennyLane's new 
+    **experimental** decomposition algorithm (by default, this new system is *not* 
+    enabled). This new system uses a graph-based approach, which provides better 
+    overall versatility and resource efficiency.
 
 Using a gate set
 ****************
 
-The example below demonstrates how a three-wire circuit can be decomposed using a pre-defined set of gates: 
+The example below demonstrates how a three-wire circuit can be decomposed using 
+a pre-defined set of gates: 
 
 .. code-block:: python
     
     from pennylane.transforms import decompose
     from functools import partial
 
+    # Works with the new decomposition system, as well.
+    # qml.decompositions.enable_graph()
```

Comment:
@isaacdevlugt this might be superfluous -- e.g., if this works with graph enabled or disabled, I wouldn't even worry about mentioning this here as it can be confusing!